### PR TITLE
[Improvement-5018][Datax] Re-assign DATAX_HOME to /opt/soft/datax instead of /opt/soft/datax/bin/datax.py

### LINKED ISSUE #5018: [Improvement][Datax] Re-assign DATAX_HOME to /opt/soft/datax instead of /opt/soft/datax/bin/datax.py
**Describe the question**

`DATAX_HOME=/opt/soft/datax/bin/datax.py` in `script/env/dolphinscheduler_env.sh` is not reasonable

Including the quick start page on the [official website](https://github.com/alibaba/DataX/blob/master/userGuid.md), `DATAX_HOME` represents the root directory of Datax. So `/opt/soft/datax` is more reasonable.

**What are the current benefits of improvement**
- Avoid misuse of `DATAX_HOME`
- Unify all HOME environment variables

**Which version of DolphinScheduler:**
 -[1.3.x]
 -[dev]

### PR DESCRIPTION
**Pull request by chengshiwen**
## What is the purpose of the pull request

*[Improvement][Datax] Re-assign `DATAX_HOME` to `/opt/soft/datax` instead of `/opt/soft/datax/bin/datax.py`*

This closes #5018

## Brief change log

  - *Re-assign `DATAX_HOME` to `/opt/soft/datax` instead of `/opt/soft/datax/bin/datax.py`*

## Verify this pull request

This pull request is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

**Comment by zhuangchong** [comment]
+1

File: docker/kubernetes/dolphinscheduler/values.yaml [hunk 1]
```diff
@@ -76,12 +76,12 @@ common:
...
     - "export SPARK_HOME2=/opt/soft/spark2"
-    #- "export PYTHON_HOME=/opt/soft/python"
+    - "export PYTHON_HOME=/usr"
     - "export JAVA_HOME=/usr/lib/jvm/java-1.8-openjdk"
     - "export HIVE_HOME=/opt/soft/hive"
     - "export FLINK_HOME=/opt/soft/flink"
-    - "export DATAX_HOME=/opt/soft/datax/bin/datax.py"
-    - "export PATH=$HADOOP_HOME/bin:$SPARK_HOME1/bin:$SPARK_HOME2/bin:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$PATH"
+    - "export DATAX_HOME=/opt/soft/datax"
+    - "export PATH=$HADOOP_HOME/bin:$SPARK_HOME1/bin:$SPARK_HOME2/bin:$PYTHON_HOME/bin:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$DATAX_HOME/bin:$PATH"
```
Comment (wen-hemin) [inline docker/kubernetes/dolphinscheduler/values.yaml:84]: There is no need to add to path.

File: docker/docker-swarm/docker-stack.yml [hunk 2]
```diff
@@ -171,11 +171,11 @@ services:
       HADOOP_CONF_DIR: "/opt/soft/hadoop/etc/hadoop"
       SPARK_HOME1: "/opt/soft/spark1"
       SPARK_HOME2: "/opt/soft/spark2"
-      #PYTHON_HOME: "/opt/soft/python"
+      PYTHON_HOME: "/usr"
```
Comment (wen-hemin) [inline docker/docker-swarm/docker-stack.yml]: This is not within the scope of this change and no modification is recommended.

File: docker/docker-swarm/docker-compose.yml [hunk 3]
```diff
@@ -177,11 +177,11 @@ services:
       HADOOP_CONF_DIR: "/opt/soft/hadoop/etc/hadoop"
       SPARK_HOME1: "/opt/soft/spark1"
       SPARK_HOME2: "/opt/soft/spark2"
-      #PYTHON_HOME: "/opt/soft/python"
+      PYTHON_HOME: "/usr"
```
Comment (wen-hemin) [inline docker/docker-swarm/docker-compose.yml]: This is not within the scope of this change and no modification is recommended.

File: docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl [hunk 4]
```diff
@@ -25,4 +25,4 @@ export HIVE_HOME=$HIVE_HOME
 export FLINK_HOME=$FLINK_HOME
 export DATAX_HOME=$DATAX_HOME
 
-export PATH=$HADOOP_HOME/bin:$SPARK_HOME1/bin:$SPARK_HOME2/bin:$PYTHON_HOME/bin:$JAVA_HOME/bin:$HIVE_HOME/bin:$PATH:$FLINK_HOME/bin:$DATAX_HOME:$PATH
+export PATH=$HADOOP_HOME/bin:$SPARK_HOME1/bin:$SPARK_HOME2/bin:$PYTHON_HOME/bin:$JAVA_HOME/bin:$HIVE_HOME/bin:$FLINK_HOME/bin:$DATAX_HOME/bin:$PATH
```
Comment (wen-hemin) [inline docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl:28]: There is no need to add to path.

**Review by wen-hemin — CHANGES_REQUESTED** [review]
There are some suggestions

File: docker/kubernetes/dolphinscheduler/values.yaml (hunk 1 above)
Comment (chengshiwen) [reply docker/kubernetes/dolphinscheduler/values.yaml:84]: The semantics of `PYTHON_HOME` has changed. In order to ensure that docker/k8s can execute python task normally, I need to make this modification. At the same time, just keep consistency with other `dolphinscheduler_env.sh`

File: docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl (hunk 4 above)
Comment (chengshiwen) [reply docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl:28]: The semantics of `PYTHON_HOME` has changed. In order to ensure that docker/k8s can run normally, I need to make this modification

Comment (chengshiwen) [reply docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl:28]: I think there is no problem with this line

File: docker/docker-swarm/docker-compose.yml (hunk 3 above)
Comment (chengshiwen) [reply docker/docker-swarm/docker-compose.yml]: Done

File: docker/docker-swarm/docker-stack.yml (hunk 2 above)
Comment (chengshiwen) [reply docker/docker-swarm/docker-stack.yml]: Done

File: docker/kubernetes/dolphinscheduler/values.yaml (hunk 1 above)
Comment (wen-hemin) [reply docker/kubernetes/dolphinscheduler/values.yaml:84]: Yes, DATAX_HOME is no need

File: docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl (hunk 4 above)
Comment (wen-hemin) [reply docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl:28]: Yes, DATAX_HOME is no need

File: docker/kubernetes/dolphinscheduler/values.yaml (hunk 1 above)
Comment (chengshiwen) [reply docker/kubernetes/dolphinscheduler/values.yaml:84]: Just keep consistency with other `dolphinscheduler_env.sh`. In fact, none of these environment variables such as `SPARK_HOME1`, `SPARK_HOME2`, `PYTHON_HOME` and `DATAX_HOME` in `PATH` are needed. In future, I will give a refactoring for the `HOME` environment variables, please refer to #5035. We don’t need the PATH environment variable at all.

File: docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl (hunk 4 above)
Comment (chengshiwen) [reply docker/build/conf/dolphinscheduler/env/dolphinscheduler_env.sh.tpl:28]: Just keep consistency with other `dolphinscheduler_env.sh`. In fact, none of these environment variables such as `SPARK_HOME1`, `SPARK_HOME2`, `PYTHON_HOME` and `DATAX_HOME` in `PATH` are needed. In future, I will give a refactoring for the `HOME` environment variables, please refer to #5035. We don’t need the PATH environment variable at all.

File: docker/kubernetes/dolphinscheduler/values.yaml (hunk 1 above)
Comment (wen-hemin) [reply docker/kubernetes/dolphinscheduler/values.yaml:84]: OK

**Review by wen-hemin — APPROVED** [review]
+1
Good job.

**Pull request merged**